packages/design-system-web/src/components/geov-yasgui/geov-yasgui.tsx

```diff
@@ -84,7 +83,9 @@ GROUP BY ?label ?long ?lat ?type ?link
     },
   ];
 
-  @Prop() defaultPlugin: 'map' | 'response' | 'table' = 'map';
+  @Prop() plugins: any[] = [];
```

Comment:
I couldn't manage to add the type here. If I put Plugin<PluginConfig> here, then some properties are still missing...

Do you know how to properly set the type here? @joschne

Reply:
When a user adds our web components library (e.g from CDN) to their website, it will initially only load a very minimal loader script. This [loader script lazy-loads web components](https://stenciljs.com/docs/distribution) if needed.

In this sense I think we can add a @Prop() plugins: string[] with the names of the plugins the user wants to enable.

In the setup of the component we can then register only these plugins (pseudo code):

```javascript
if(this.plugins.includes('map')  this.Y.Yasr.registerPlugin('map', MyMapPlugin.default as any);
```

The MyMapPlugin will be responsible to load the web component, as here `geov-map-one`:

```typescript
  // Draw the result set. This plugin creates a <geov-map-one/ data=[...]>
  draw() {
    const el = document.createElement('geov-map-one');
    el.data = this.yasr.results.getBindings();
    this.yasr.resultsEl.appendChild(el);
  }
```

This will then trigger the lazy loading of the geov-map-one component.

The only "overhead" added for unused plugins are the classes like MyMapPlugin. I think, this is no problem.

What do you think @flicksolutions

Reply:
so, for each new plugin we create, we create a new line
```
if(this.plugins.includes('x')  this.Y.Yasr.registerPlugin('x', MyXPlugin.default as any);
```
?

That should also work, yeah!